lib/CL/pocl_cl.h

```diff
@@ -1718,6 +1734,16 @@ struct _cl_kernel {
      will be synchronized to the device. */
   char can_access_all_raw_buffers_indirectly;
 
+  /* Custom kernel runner (a host function). Used for kernels not
```

Comment:
This breaks the framework design. The execution related part belongs fully to the device driver. We can and will have a number of different devices in the platform which each have their own implementation of the same DBK, each implementing them behind the scenes of the driver layer (lib/devices).